### PR TITLE
Update Laravel Tutorial

### PR DESCRIPTION
**Pull request by jobannon**
# Summary
Updates to the laravel tutorial based on a recent walkthrough.

### CODE REVIEW

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
 ---
 layout: blog-post
 title: User registration and sign-in with Laravel and FusionAuth
-description: We'll build a Laravel application that delegates user management, registration and login to FusionAuth.
+description: We'll build a Laravel application that delegates user management, registration, and login to FusionAuth.
 author: Karl Hughes
 image: blogs/fusionauth-laravel/user-registration-sign-in-laravel-fusionauth.png
 category: blog
-tags: client-php
+updated-date: 2020-30-03
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:8]: It's `updated_date`, not `updated-date`, I believe.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+tags: client-php, tutorial, tutorial-laravel
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:9]: Please add `tutorial-php` as well. Probably should add `tutorial-javascript` to the vue blog post too. My bad for forgetting.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 3]
```diff
@@ -46,6 +47,10 @@ composer create-project --prefer-dist laravel/laravel fusionauth-laravel
 
 There are several other options for [installing](https://laravel.com/docs/7.x#installation) and [configuring](https://laravel.com/docs/7.x#configuration) Laravel, so be sure to check out the documentation. Once you've installed Laravel, you can use [Homestead](https://laravel.com/docs/7.x/homestead), [Valet](https://laravel.com/docs/7.x/valet), [Docker](https://www.digitalocean.com/community/tutorials/how-to-set-up-laravel-nginx-and-mysql-with-docker-compose), or [PHP's built-in webserver](https://scotch.io/tutorials/prototype-quickly-in-laravel-with-phps-built-in-server-and-sqlite) to run it locally.
 
+{% include _callout-tip.liquid content=
+"One simple option to run your site locally is the command `php artisan serve`"
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:51]: :boom: awesome!

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 4]
```diff
@@ -64,6 +69,10 @@ FUSIONAUTH_BASE_URL=http://localhost:9011
 Finally, you can take advantage of [Laravel's dependency injection system](https://laravel.com/docs/7.x/providers) by registering the `FusionAuthClient` in the `boot()` method of your `app/Providers/AppServiceProvider.php` file:
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:69]: Did we switch versions of Laravel? Which version did you test against. Might want to note that here.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 5]
```diff
@@ -178,23 +187,44 @@ class RegisterUser extends Controller
     return redirect('/');
   }
 }
+//..
 ```
 
-_Note: FusionAuth includes several other options for user registration. Be sure to [check out the documentation](https://fusionauth.io/docs/v1/tech/apis/registrations) to make sure you include all the appropriate options._
+Finally, you'll need to create two routes to register new users: one to display the registration form and another to process the data and interact with the FusionAuth client. Add the following lines to your `routes/web.php` file (remove the welcome route):
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:193]: Curious why you moved the routes down. Did that just make more sense?

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 6]
```diff
@@ -178,23 +187,44 @@ class RegisterUser extends Controller
...
+
+```php
+Route::view('/register', 'register');
+Route::post('/register', 'RegisterUser');
+```
+
+{% include _callout-note.liquid content=
+"FusionAuth includes several other options for user registration. Be sure to [check out the documentation](https://fusionauth.io/docs/v1/tech/apis/registrations) to make sure you include all the appropriate options."
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:201]: Maybe `FusionAuth supports different features when a user registers, including verification emails, required fields and custom roles. Be sure to ...`

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 7]
```diff
@@ -178,23 +187,44 @@ class RegisterUser extends Controller
...
 
 At this point, you should be able to access the `/register` route in your Laravel application and successfully register a new user in FusionAuth.
 
-{% include _image.liquid src="/assets/img/blogs/fusionauth-laravel/fusionauth-2020-05-22-a.png" alt="The registration page for a Laravel application using FusionAuth." class="img-fluid" figure=false %}
+{% include _image.liquid src="/assets/img/blogs/fusionauth-laravel/new-user-register.png" alt="The registration page for a Laravel application using FusionAuth." class="img-fluid bottom-cropped" figure=false %}
 
-## Logging a user in
-Now that users can register a new account, you'll need to allow them to login using the email and password they created during registration. This will require two new routes, so open your `routes/web.php` file again and add the following lines:
+{% include _callout-tip.liquid content=
+"Getting errors at this point?  One common mistake is forgetting to import the FusionAuth client into your controller - `use FusionAuth\FusionAuthClient;`"
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:209]: :boom:

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 8]
```diff
@@ -178,23 +187,44 @@ class RegisterUser extends Controller
...
+%}
 
-```php
-Route::view('/', 'login');
-Route::post('/login', 'LoginUser');
-```
+##  Let's take a look at our progress
+
+Previous to the registration step above, no new users had been added or registered yet.
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:214]: Maybe `Before we built out and called the registration functionality above, no new users...`.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 9]
```diff
@@ -178,23 +187,44 @@ class RegisterUser extends Controller
...
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-laravel/no-user-registration.png" alt="No users in our application yet!" class="img-fluid" figure=false %}
+
+Our newly registered user!
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:218]: You could ham this up even more:

```
Behold our newly registered user! Bask in the warmth of your registration coding abilities.
```

Or similar.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 10]
```diff
@@ -218,7 +248,7 @@ The login form is very similar to the registration form created in the previous
 @endsection
 ```
 
-Finally, to process user logins, you'll need to create another single action controller. First, run the artisan command:
+To process user logins, you'll need to create another single action controller. First, run the artisan command:
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:251]: Is artisan a cli tool? If so, \`artisan\` is better.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 11]
```diff
@@ -257,11 +292,26 @@ class LoginUser extends Controller
...
 
-_Note: When users are logged in, FusionAuth also returns a JWT token. If you're building a single page application or want to perform client-side requests, [check out the documentation](https://fusionauth.io/docs/v1/tech/apis/jwt) for details._
+The first route will display a new view file that includes an email and password input field. The second route will use a new action controller to log a user in via FusionAuth. Your login page should look like this:
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-laravel/login-page.png" alt="The profile page for a Laravel application using FusionAuth." class="img-fluid bottom-cropped" figure=false %}
+
+{% include _callout-tip.liquid content="When users are logged in, FusionAuth also returns a JWT token. If you're building a single page application or want to perform client-side requests, [check out the documentation](https://fusionauth.io/docs/v1/tech/apis/jwt) for details."
+%}
+
+Now when you login using the new form, you'll be redirected to `/profile`, but that page doesn't exist yet (hence, the 404 error)
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:312]: Missing a period on the end of this sentence.

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 12]
```diff
@@ -311,28 +361,28 @@ Now you can create a `profile.blade.php` file in your `resources/views/` directo
 ```
 {% endraw %}
 
-Any other user data you choose to store in FusionAuth will be available in the `session()->get('user')` array if you'd like to show it to logged in users.
+Any other user data you choose to store in FusionAuth will be available in the `session()->get('user')` array if you'd like to show it to logged-in users.
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:364]: I think we use `logged in` (no dash). A search doesn't show any `logged-in` terms. Can you please add that to the devdocs?

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md [hunk 13]
```diff
@@ -311,28 +361,28 @@ Now you can create a `profile.blade.php` file in your `resources/views/` directo
...
 
 When you log in, you should see a profile page that looks something like this:
 
-{% include _image.liquid src="/assets/img/blogs/fusionauth-laravel/fusionauth-2020-05-22-b.png" alt="The profile page for a Laravel application using FusionAuth." class="img-fluid" figure=false %}
+{% include _image.liquid src="/assets/img/blogs/fusionauth-laravel/profile-user-view.png" alt="The profile page for a Laravel application using FusionAuth." class="img-fluid bottom-cropped" figure=false %}
```
Comment (mooreds) [inline site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:368]: Thanks for renaming the images!

Comment (mooreds) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:368]: Does bottom-cropped do anything? I thought that was only in the docs css.

**Review by mooreds — CHANGES_REQUESTED** [review]

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 2 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:9]: Makes sense!

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 4 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:69]: Laravel Framework 7.30.4 
- I will add a note to the GH repo too 👍🏻

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 7 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:209]: 🤜🏻 🤛🏻

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 10 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:251]: `php artisan` gives you a subset of Laravel commands - will adjust markdown 👍🏻

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 11 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:312]: the `period` is `404` just like our profile page ... lol 🤣 
...and be happy that my calling was not comedy 😄

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 12 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:364]: Good ole Grammarly suggested that `-`.  Will update devDocs 👍🏻

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 13 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:368]: It looks it does not (I assumed that it added margin, but that is already part of the `img` attribute).  I will remove the unneeded classes (looks like we have margin to spare) 👍🏻

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 9 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:218]: nice!

File: site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md (hunk 5 above)
Comment (jobannon) [reply site/_posts/2020-06-03-user-registration-and-sign-in-with-laravel.md:193]: If the routes are out of order you will get an error....  
You need to run 

```shell
php artisan make:controller RegisterUser --invokable
```
before you add the route in `web.php`

I will another `important` tip!

**Comment by jobannon** [comment]
@mooreds 
Thanks for the review!  Will give this a once over Monday and then it should be ready to ship! 🚀

**Pull request merged**